src/liballoc/collections/binary_heap.rs

```diff
@@ -165,6 +165,9 @@ use super::SpecExtend;
 /// trait, changes while it is in the heap. This is normally only possible
 /// through `Cell`, `RefCell`, global state, I/O, or unsafe code.
 ///
+/// Both `push` and `pop` operations can be performed in `O(log(n))` time,
```

Comment:
I think this would need to call out that the O given for `push` is an *amortized* cost, similar to how this is emphasized in the std::collections doc.

Reply:
Thanks for your comment! I've read through `std::collections` once more and it basically means that `push` can re-allocate and therefore would be O(n) right? What would you think about a change like this?:

> The amortized cost of a `push` operation is `O(log(n))`. When the buffer cannot hold more elements a resize costs `O(n)`, since all elements have to be copied to a new memory region. `pop` operations never reallocate and can be performed in `O(log(n))` time, whereas `peek` can be performed in `O(1)` time.

Reply:
Sure, that would work. I would maybe de-emphasize the O(n) statement slightly because that figure would rarely be relevant to selecting a data structure. Notice how std::collections treats it almost like a sidenote explaining what amortized cost refers to, as separate from the more meaningful comparable quantities in the tables.

Reply:
To clarify, in case my two comments look like they contradict each other in terms of what is worth emphasizing:

The fact that O(log(n)) is an amortized cost rather than worst case is important (which is why std::collections places `*` markers directly alongside the important data) but the O(n) worst case time may be treated as a less important detail (which is why std::collections has that only in a sidenote).

Reply:
@dtolnay I changed the PR! Hopefully, this is a bit better :)